### PR TITLE
Docs: Fix spelling mistakes

### PR DESCRIPTION
**Pull request by bmish**
Searched using cspell. I manually fixed the obvious typos while ignoring thousands of false positives.

#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).
- [ ] The team has reached consensus on the changes proposed in this pull request. If not, I understand that the evaluation process will begin with this pull request and won't be merged until the team has reached consensus.

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

I searched for typos / spelling mistake using [cspell](https://www.npmjs.com/package/cspell). I manually fixed the obvious typos that were reported (and ignored thousands of other false positive matches).

#### Is there anything you'd like reviewers to focus on?

No.

### CODE REVIEW

**Comment by mdjermanovic** [comment]
Thanks for fixing all these!

I just changed the title to `Docs:` because some of the changes are in the user-facing documentation.

File: docs/rules/no-fallthrough.md [hunk 1]
```diff
@@ -49,7 +49,7 @@ switch(foo) {
 switch(foo) {
     case 1:
         doSomething();
-        // fallsthrough
+        // falls through
```
[the remark this answers is not in the document]
Comment (bmish) [reply docs/rules/no-fallthrough.md]: Good point, reverted this change.

[the remark this answers is not in the document]
Comment (mdjermanovic) [reply docs/rules/no-fallthrough.md]: Thanks, that was really fast! :+1: 

Other changes look good to me, though I'd like to check everything once again (probably tomorrow).

**Comment by kaicataldo** [comment]
Thanks for working on this! I think it's probably best not to change the three test files listed above by @mdjermanovic. Otherwise, this LGTM!

**Comment by bmish** [comment]
Thanks for reviewing. I removed the changes to those 3 files.

File: lib/rules/no-unused-vars.js [hunk 2]
```diff
@@ -282,7 +282,7 @@ module.exports = {
         function getRhsNode(ref, prevRhsNode) {
             const id = ref.identifier;
             const parent = id.parent;
-            const granpa = parent.parent;
+            const grandpa = parent.parent;
```
Comment (kaicataldo) [inline lib/rules/no-unused-vars.js]: Can we change this to `grandparent`?

Comment (bmish) [reply lib/rules/no-unused-vars.js]: Done!

Comment (kaicataldo) [reply lib/rules/no-unused-vars.js]: Thank you!

**Comment by kaicataldo** [comment]
Thanks for contributing to ESLint!

**Pull request merged**